src/main/kotlin/minesweeper/domain/Field.kt

```diff
@@ -1,64 +1,65 @@
 package minesweeper.domain
 
 class Field(
-    private val fieldInfo: FieldInfo,
-    private val mineCount: MineCount,
-    private val spotGenerator: SpotGenerator,
+    val fieldInfo: FieldInfo,
+    private val minePositions: Set<Position>,
 ) {
-    private val width = fieldInfo.getWidth()
-    val lines: List<FieldLine> = createField()
+    private val spots: Map<Position, Spot> = createField()
 
     init {
         validateMineCount()
     }
 
-    private fun createField(): List<FieldLine> {
-        val spots = spotGenerator.generate(fieldInfo, mineCount)
-        return spots.mapIndexed { index, spot ->
-            if (spot is SafeSpot) {
-                val y = index / width
-                val x = index % width
-                val nearbyMineCount = countAdjacentMines(spots, y, x)
-                spot.updateNearbyMineCount(nearbyMineCount)
+    private fun createField(): Map<Position, Spot> {
+        return (0 until fieldInfo.getWidth() + 1).flatMap { x ->
+            (0 until fieldInfo.getHeight() + 1).map { y ->
+                Position(x, y)
+            }
+        }.associateWith { position ->
+            when {
+                minePositions.contains(position) -> MineSpot(position)
+                else -> SafeSpot(position)
             }
-            spot
-        }.chunked(width).map { lineSpots ->
-            FieldLine(lineSpots)
         }
     }
 
-    private fun countAdjacentMines(
-        spots: List<Spot>,
-        y: Int,
-        x: Int,
-    ): Int {
-        return NEARBY.count { (dy, dx) ->
-            val newY = y + dy
-            val newX = x + dx
-            isWithinBounds(newY, newX) && spots[newY * width + newX].isMine()
-        }
+    fun getSpot(position: Position): Spot {
+        return spots[position] ?: throw IllegalArgumentException("해당 위치에 대한 Spot이 존재하지 않습니다.")
     }
 
-    private fun isWithinBounds(
-        y: Int,
-        x: Int,
-    ): Boolean {
-        return y in 0 until fieldInfo.getHeight() && x in 0 until width
+    private fun validateMineCount() {
+        val totalPossibleSpots = fieldInfo.getHeight() * fieldInfo.getWidth()
+        require(minePositions.size <= totalPossibleSpots) { "지뢰 개수는 필드의 총 스팟보다 많을 수 없습니다." }
     }
 
-    private fun validateMineCount() {
-        val height = fieldInfo.getHeight()
-        val totalSpots = height * width
-        require(mineCount.count <= totalSpots) { "지뢰 개수는 필드의 총 스팟보다 많을 수 없습니다." }
+    fun openSpot(position: Position): OpenResult {
+        val targetSpot =
+            spots[position]?.let {
+                if (it.isMine()) {
+                    return OpenResult.GameOver
+                }
+                it
+            } as SafeSpot
```

Comment:
kotlin의 let 스코프 함수를 활용해볼 수도 있지만, elvis operator를 활용해보는 방법도 좋아보여요!
```suggestion
        val targetSpot = spots[position] ?: return OpenResult.GameOver
```